pkg/credentials/grpc.go

```diff
@@ -19,20 +19,19 @@ func GetServerOptions(certChain *CertChain) ([]grpc.ServerOption, error) {
 	cp := x509.NewCertPool()
 	cp.AppendCertsFromPEM(certChain.RootCA)
 
-	if certChain != nil {
-		cert, err := tls.X509KeyPair(certChain.Cert, certChain.Key)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create server certificate: %s", err)
-		}
+	cert, err := tls.X509KeyPair(certChain.Cert, certChain.Key)
+	if err != nil {
```

Comment:
This shouldn't return an error here, as users can run Dapr with mTLs disabled, in which case this should return the options without the TLS configuration.

Reply:
I only removed the unnessasory if check of "if certChain != nil {...}"

Do you mean that the check of this "err != nil" should return  an empty opts:

```go
cert, err := tls.X509KeyPair(certChain.Cert, certChain.Key)
		if err != nil {
			// return nil, fmt.Errorf("failed to create server certificate: %s", err)
                          return opts, nil
		}
```

OK, I will do this update.